packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -740,10 +748,10 @@ export const createAssetNodes = ({
       },
       url: `https:${file.url}`,
       placeholderUrl: `https:${file.url}?w=%width%&h=%height%`,
-      mimeType: file.contentType,
-      filename: file.fileName,
-      width: file.details?.image?.width,
-      height: file.details?.image?.height,
+      mimeType: file.contentType ?? null,
+      filename: file.fileName ?? null,
```

Comment:
These two props can't be null
```suggestion
      mimeType: file.contentType,
      filename: file.fileName,
```

Reply:
@wardpeet Unfortunately they can in some cases when using the Preview API and a (big) asset is still processed by Contentful

I talked to the API team about it. So I added another check for `file.url` to exist and more comments to this section.

Reply:
we can't create an asset when these things are null cause Image CDN needs them.

Reply:
alright, then lets require them as well! :)